spiceypy/spiceypy.py

```diff
@@ -12668,7 +12734,7 @@ def srfxpt(method, target, et, abcorr, obsrvr, dref, dvec):
     if hasattr(et, "__iter__"):
         return numpy.array(
                 [srfxpt(method, target, t, abcorr, obsrvr, dref, dvec) for t in
-                 et])
+                 et]), True  # Need trailing True to spoof @spiceError* above
```

Comment:
(And you left off the "good catch" too ;-)

I spent a lot of time spinning my wheels on this.  As we found out, I didn't know the "decorator" jargon when you mentioned it, but I had already figured out what the @ stuff was doing functionally.

Also, I found this to be an odd result, because I thought the @spiceFoundExceptionThrower decorator checks the type of last item of the return (res[-1]) and should do nothing it that last item is not a [True/False] bool.  So summat odd is happening there - no wait, it _only_ checks type for the found==False case, so this is a bug, and the decorator should use `elif isinstance(found,bool)` instead of just `else`.

Wait a minute:  couldn't we change
`if isinstance(found, bool) and not found:` and `else:`
to simply
`if found is False:` and `elif found is True:`
in the spiceFoundExceptionThrower decorator?

Wouldn't a better (in the sense of more robust) decorator be
```
def spiceFoundExceptionThrower(f)
    @functools.wrapps(f)
    def wrapper(*args,**kwargs):
        res = f(*args,**kwargs):
        if not isinstance(res,tuple):   ### <== the A.R. side of me wants to do this
            return res
        found = res[-1]                 ### <== aack!
        if found is False:
            raise ...
        elif found is True:
            if len(res) == 2:
                return res[0]
            else:
                return res[:-1]
        return res
```
I'm still bothered by the possibility that the highlighted line (cf. [aack!] above) could throw an error (`found = ((None,)+res[-1:])[-1]`) would be more robust), but presumably that is taken care of by the choice of decorated functions.